### PR TITLE
refactor(opentracing): add tests and pass correct span contexts

### PR DESCRIPTION
**Pull request by vigneshshanmugam**
+ Fix the issue when the parent span(handle-request) was getting finished before the child span(fetch-fragment) is finished. 
+ Add tests for the opentracing instrumentation using [MockTracer](https://opentracing-javascript.surge.sh/classes/mocktracer.html)

### CODE REVIEW

File: lib/fragment.js [hunk 1]
```diff
@@ -97,43 +97,41 @@ module.exports = class Fragment extends EventEmitter {
         const spanOptions = parentSpan ? { childOf: parentSpan } : {};
         const span = tracer.startSpan('fetch-fragment', spanOptions);
         span.addTags({
-            'fragment-url': url,
-            'fragment-id': this.attributes.id || 'unnamed',
+            [Tags.SPAN_KIND]: Tags.SPAN_KIND_RPC_CLIENT,
+            url: url,
```
Comment (vigneshshanmugam) [inline lib/fragment.js]: we know this is a fragment span from operation name so removed the `fragment`

**Comment by lmineiro** [comment]
:+1:

**Comment by vigneshshanmugam** [comment]
👍

**Comment by lmineiro** [comment]
:+1:

**Pull request merged**